test/integration/index-management/search-index-management.test.ts

```diff
@@ -21,64 +21,80 @@ describe('Search Index Management Integration Tests', function () {
+          }
+        },
+        test: async function () {
+          let res;
+          await client.withSession(async session => {
+            await session
+              .withTransaction(
+                async function (session) {
+                  res = collection.listSearchIndexes({ session });
+                  await res.toArray();
+                },
+                { readConcern: 'local', writeConcern: { w: 1 } }
+              )
+              .catch(e => e);
+            // expect(e.errmsg).to.match(/^.*Atlas.*$/) - uncomment after NODE-6047
+            expect(commandStartedEvents[0]).to.exist;
+            // flip assertion after NODE-6047 implementation
+            expect(commandStartedEvents[0]?.command?.readConcern).to.exist;
+            expect(commandStartedEvents[0]?.command?.writeConcern).to.not.exist;
+          });
+        }
+      });
     });
 
     context('when listSearchIndexes operation is run with causalConsistency', function () {
-      it.skip('should not include write concern or read concern in command', async function () {
-        await client.withSession({ causalConsistency: true }, async session => {
-          try {
+      it('should not include write concern or read concern in command', {
+        metadata: {
+          requires: {
+            topology: '!single',
+            mongodb: '>=7.0',
+            serverless: 'forbid'
+          }
+        },
+        test: async function () {
+          await client.withSession({ causalConsistency: true }, async session => {
             const res = collection.listSearchIndexes({ session });
-            await res.toArray();
-          } catch (e) {
-            expect(e.errmsg).to.match(/^.*Atlas.*$/);
-          } finally {
+            await res.toArray().catch(e => expect(e.errmsg).to.match(/^.*Atlas.*$/));
             expect(commandStartedEvents[0]).to.exist;
-            expect(commandStartedEvents[0].command.readConcern).to.not.exist;
-            expect(commandStartedEvents[0].command.writeConcern).to.not.exist;
-          }
-        });
-      }).skipReason =
-        'TODO(NODE-6047): Ignore read/write concern in applySession for Atlas Search Index Helpers';
+            expect(commandStartedEvents[0]?.command?.readConcern).to.not.exist;
+            expect(commandStartedEvents[0]?.command?.writeConcern).to.not.exist;
+          });
+        }
+      });
     });
 
     context('when listSearchIndexes operation is run with snapshot on', function () {
-      it.skip('should not include write concern or read concern in command', async function () {
-        await client.withSession({ snapshot: true }, async session => {
-          try {
+      // TODO(NODE-6047): Ignore read/write concern in applySession for Atlas Search Index Helpers
+      it('should include write concern or read concern in command - TODO(NODE-6047)', {
+        metadata: {
+          requires: {
+            topology: ['replicaset', 'sharded'],
```

Comment:
Load-balanced is excluded from topologies. This is because Mongodb versioning for snapshots is not captured by load-balanced test variants. For example, a rapid test will say, "snapshot requires MongoDB >= 5.0." Here's a [link to an example](https://spruce.mongodb.com/task/mongo_node_driver_next_rhel80_large_Node20_test_rapid_load_balanced_patch_abf8bdfc8dbae85356a7cef9d476e619a79e53d9_660f196e6650da000703866d_24_04_04_21_20_47/logs?execution=0&sortBy=STATUS&sortDir=ASC).